scheduler/pkg/agent/repository/mlserver/settings.go

```diff
@@ -0,0 +1,182 @@
+package mlserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/seldonio/seldon-core/apis/go/v2/mlops/scheduler"
+)
+
+var (
+	sklearnExtensions  = []string{"joblib", "pkl", "pickle"}
```

Comment:
given that mlserver is [very strict](https://github.com/SeldonIO/MLServer/blob/3a81135aa1da96cc0b0c5748ce2ad7ad540c0d3b/runtimes/sklearn/mlserver_sklearn/sklearn.py#L23) about the full name of the model file, should we also try to match here with `model.joblib` etc?

Reply:
I think this only applies if the URI provided in the model-settings.json is a folder. In this case as we are working backwards and creating the model-settings.json we can be more lenient. see [here](https://github.com/SeldonIO/MLServer/blob/3a81135aa1da96cc0b0c5748ce2ad7ad540c0d3b/mlserver/utils.py#L30).